spec.html

```diff
@@ -17760,6 +17779,121 @@ <h1>EnumerateObjectProperties ( _O_ )</h1>
+                    1. If Type(_key_) is String, then
+                      1. Append _key_ to _remaining_.
+                  1. Set _O_.[[ObjectWasVisited]] to *true*.
+                1. Repeat, while _remaining_ is not empty,
+                  1. Remove the first element from _remaining_ and let _r_ be the value of the element.
+                  1. If there does not exist an element _v_ of _visited_ such that SameValue(_r_, _v_) is *true*, then
+                    1. Let _desc_ be ? _object_.[[GetOwnProperty]](_r_).
+                    1. If _desc_ is not *undefined*, then
+                      1. Append _r_ to _visited_.
+                      1. If _desc_.[[Enumerable]] is *true*, return CreateIterResultObject(_r_, *false*).
+                1. Set _object_ to ? _object_.[[GetPrototypeOf]]().
+                1. Set _O_.[[Object]] to _object_.
+                1. Set _O_.[[ObjectWasVisited]] to *false*.
+                1. If _object_ is *null*, return CreateIterResultObject(*undefined*, *true*).
+            </emu-alg>
+          </emu-clause>
+        </emu-clause>
+
+        <emu-clause id="sec-properties-of-for-in-iterator-instances">
+          <h1>Properties of For In Iterator Instances</h1>
+          <p>For In Iterator instances are ordinary objects that inherit properties from the %ForInIteratorPrototype% intrinsic object. For In Iterator instances are initially created with the internal slots listed in <emu-xref href="#table-for-in-iterator-instance-slots"></emu-xref>.</p>
+              <emu-table id="table-for-in-iterator-instance-slots" caption="Internal Slots of For In Iterator Instances">
+            <table>
+              <tbody>
+              <tr>
+                <th>
+                  Internal Slot
+                </th>
+                <th>
+                  Description
+                </th>
+              </tr>
+              <tr>
+                <td>
+                  [[Object]]
+                </td>
+                <td>
+                  The Object value whose properties are being iterated.
+                </td>
+              </tr>
+              <tr>
+                <td>
+                  [[ObjectWasVisited]]
+                </td>
+                <td>
+                  *true* if the iterator has invoked [[OwnPropertyKeys]] on [[Object]], *false* otherwise.
+                </td>
+              </tr>
+              <tr>
+                <td>
+                  [[VisitedKeys]]
+                </td>
+                <td>
+                  A list of String values which have been emitted by this iterator thus far.
+                </td>
+              </tr>
+              <tr>
+                <td>
+                  [[RemainingKeys]]
+                </td>
+                <td>
+                  A list of String values remaining to be emitted for the current object, before iterating those on its prototype (if its prototype is not *null*).
```

Comment:
"before iterating the properties of its prototype"? The antecedent of "those" reads like it's String values, which could be clearer.